Marlin/src/HAL/AVR/fast_pwm.cpp

```diff
@@ -66,11 +66,11 @@ const Timer get_pwm_timer(const pin_t pin) {
       case TIMER1A: case TIMER1B:
     #endif
 
-    break;    // Protect reserved timers (TIMER0 & TIMER1)
+    break;    // Protect reserved timers (TIMER0A & TIMER1)
 
     #ifdef TCCR0A
       case TIMER0B:   // Protected timer, but allow setting the duty cycle on OCR0B for pin D4 only
-        return Timer({ { &TCCR0A, nullptr, nullptr }, { (uint16_t*)&OCR0B, nullptr, nullptr }, nullptr, 0, 0, true, true });
+        return Timer({ { &TCCR0A, nullptr, nullptr }, { nullptr, (uint16_t*)&OCR0B, nullptr }, nullptr, 0, 1, true, true });
```

Comment:
It seems appropriate to move the OCR entry to the 'B' slot and set `q` to 1 (B) so that the `_SET_COMnQ` macro will affect the correct bits. And, potentially this allows timer 0 to not be a special case. Except, I note below that the `protected` state is ignored in `set_pwm_duty`.

Reply:
Yes, if used it must be corrected.